sdk/component/internal/storer/local/sqlite/sqlite.go

```diff
@@ -50,31 +77,174 @@ func NewManager(dbPath string, dbName string) (*manager, error) {
 		return nil, fmt.Errorf("could not open sqlite db: %w", err)
 	}
 
-	return &manager{
+	mgr := &manager{
+		clock:  clockwork.NewRealClock(),
 		db:     db,
 		dbName: dbName,
-	}, nil
+	}
+
+	for _, opt := range opts {
+		if err := opt(mgr); err != nil {
+			return nil, fmt.Errorf("could not apply option: %w", err)
+		}
+	}
+
+	return mgr, nil
 }
 
-// TODO - implement me.
-func (m *manager) Read(ctx context.Context) ([]*ocsf.VulnerabilityFinding, error) {
-	return nil, nil
+// Validate. TODO - implement.
+func (m *manager) Validate(*ocsf.VulnerabilityFinding) error {
+	return nil
 }
 
-// TODO - implement me.
-func (m *manager) Write(ctx context.Context, findings []*ocsf.VulnerabilityFinding) error {
+// Read finds Vulnerability Findings by workflowID.
+// It returns storer.ErrNoFindingsFound is not vulnerabilities were found.
+func (m *manager) Read(ctx context.Context, workflowID uuid.UUID) ([]*ocsf.VulnerabilityFinding, error) {
+	stmt, err := m.db.PrepareContext(ctx, `
+		SELECT (findings) 
+			FROM finding 
+			WHERE workflow_id = :workflow_id
+		;
+	`)
+	if err != nil {
+		return nil, fmt.Errorf("could not prepare select statement: %w", err)
+	}
+
+	var jsonFindingsStr string
+	if err := stmt.
```

Comment:
I'm not a huge fan of this format because the err is checked 4 lines down and it's kinda lost. I'd rather have the err checked in a separate if statement